packages/web/src/common/store/tipping/sagas.ts

```diff
@@ -263,7 +267,171 @@ function* confirmTipIndexed({
   return false
 }
 
+function* wormholeAudioIfNecessary({ amount }: { amount: number }) {
+  const walletClient = yield* getContext('walletClient')
+
+  const waudioBalanceWei = yield* call([
+    walletClient,
+    'getCurrentWAudioBalance'
+  ])
+  const audioWeiAmount = new BN(AUDIO(amount).value.toString()) as BNWei
+
+  if (isNullOrUndefined(waudioBalanceWei)) {
+    throw new Error('Failed to retrieve current wAudio balance')
+  }
+
+  // If transferring spl wrapped audio and there are insufficent funds with only the
+  // user bank balance, transfer all eth AUDIO to spl wrapped audio
+  if (audioWeiAmount.gt(waudioBalanceWei)) {
+    console.info('Converting Ethereum AUDIO to Solana wAUDIO...')
+
+    // Wait for a second before showing the notice that this might take a while
+    const showConvertingMessage = yield* fork(function* () {
+      yield delay(1000)
+      yield put(convert())
+    })
+    yield call([walletClient, walletClient.transferTokensFromEthToSol])
```

Comment:
A few `yield` instead of `yield*` in here and I still can't remember if those are ever a good idea in Typescript...